From 36093b58547f47431bc984e9397e40a10d2b5af5 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Wed, 10 Jan 2018 13:36:41 -0600
Subject: [PATCH 02/10] Fix linking to zlib when cross-compiling

---
 compiler/rustc_llvm/build.rs | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/compiler/rustc_llvm/build.rs b/compiler/rustc_llvm/build.rs
index b2ff9efb41c..49537f0e6b8 100644
--- a/compiler/rustc_llvm/build.rs
+++ b/compiler/rustc_llvm/build.rs
@@ -219,10 +219,8 @@ fn main() {
     // of llvm-config, not the target that we're attempting to link.
     let mut cmd = Command::new(&llvm_config);
     cmd.arg(llvm_link_arg).arg("--libs");
-
-    if !is_crossed {
-        cmd.arg("--system-libs");
-    }
+    cmd.arg("--system-libs");
+    cmd.args(&components);
 
     // We need libkstat for getHostCPUName on SPARC builds.
     // See also: https://github.com/llvm/llvm-project/issues/64186
@@ -256,7 +254,6 @@ fn main() {
         println!("cargo:rustc-link-lib=z");
         println!("cargo:rustc-link-lib=execinfo");
     }
-    cmd.args(&components);
 
     for lib in output(&mut cmd).split_whitespace() {
         let mut is_static = false;
-- 
2.45.2

