From 333b5798b6cef02b5a154b901789a2382fc56f5f Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Wed, 10 Jan 2018 13:36:41 -0600
Subject: [PATCH 02/14] Fix linking to zlib when cross-compiling

---
 compiler/rustc_llvm/build.rs | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/compiler/rustc_llvm/build.rs b/compiler/rustc_llvm/build.rs
index b0783d75d47..fb21b7c63ef 100644
--- a/compiler/rustc_llvm/build.rs
+++ b/compiler/rustc_llvm/build.rs
@@ -236,10 +236,8 @@ fn main() {
     // of llvm-config, not the target that we're attempting to link.
     let mut cmd = Command::new(&llvm_config);
     cmd.arg(llvm_link_arg).arg("--libs");
-
-    if !is_crossed {
-        cmd.arg("--system-libs");
-    }
+    cmd.arg("--system-libs");
+    cmd.args(&components);
 
     if (target.starts_with("arm") && !target.contains("freebsd"))
         || target.starts_with("mips-")
@@ -254,7 +252,6 @@ fn main() {
     } else if target.contains("netbsd") || target.contains("haiku") || target.contains("darwin") {
         println!("cargo:rustc-link-lib=z");
     }
-    cmd.args(&components);
 
     for lib in output(&mut cmd).split_whitespace() {
         let name = if let Some(stripped) = lib.strip_prefix("-l") {
-- 
2.41.0

